normandy/base/api/relations.py

```diff
@@ -0,0 +1,57 @@
+from django.core.urlresolvers import reverse as django_reverse
+from django.conf import settings
+
+from rest_framework.relations import HyperlinkedRelatedField
+from rest_framework.reverse import preserve_builtin_query_params
+
+
+class CDNHyperlinkedRelatedField(HyperlinkedRelatedField):
+
+    def __init__(self, **kwargs):
+        # super is going to override self.reverse, so stash it and restore it
+        reverse = self.reverse
+        super().__init__(**kwargs)
+        self.reverse = reverse
+
+    def reverse(self, viewname, args=None, kwargs=None, request=None, format=None, **extra):
```

Comment:
Overriding reverse is weird; [the code says](https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/relations.py#L254) it's only a property on the class for testing purposes.

I think it's more natural to override `get_url`. You can even avoid code duplication completely by setting the `request` argument to `None` if `CDN_URL` is filled before calling the supermethod; `request` is only ever used for building absolute URLs. If you don't include it, you'll get a non-absolute URL that you can then prepend the CDN url to.

That might be a bit unclear. I think it'd look like:

```python
def get_url():  # pretend there are args here
    if settings.CDN_URL is not None:
        request = None
        url = super().get_url(args_with_modified_request)
        # Append the CDN url and you're done!
    else
        return super().get_url()
```